### PR TITLE
Add autoscale template for VMSS

### PR DESCRIPTION
**Pull request by asheniam**
Add autoscale template for VMSS

### CODE REVIEW

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+      "type": "string",
+      "defaultValue": "Standard_A1",
+      "metadata": {
+        "description": "Size of VMs in the VM Scale Set."
+      }
+    },
+    "windowsOSVersion": {
+      "type": "string",
+      "defaultValue": "2012-R2-Datacenter",
+      "allowedValues": [
```
Comment (bmoore-msft) [inline monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: 2016???

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+      "metadata": {
+        "description": "The Windows version for the VM."
+      }
+    },
+    "vmssName": {
+      "type": "string",
+      "metadata": {
+        "description": "String used as a base for naming resources. Must be 3-61 characters in length and globally unique across Azure."
+      },
+      "maxLength": 61
```
Comment (bmoore-msft) [inline monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: add minLength

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password."
+      }
+    },
+    "minimumCapacity": {
+      "type": "Int",
```
Comment (bmoore-msft) [inline monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: add min/max values for instance count, minimum and maximum capacity

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+        "description": "The maximum capacity.  Autoscale engine will ensure the instance count is not greater than this value."
+      }
+    },
+    "defaultCapacity": {
+      "type": "Int",
+      "metadata": {
+        "description": "The default capacity.  Autoscale engine will preventively set the instance count to be this value if it can not find any metric data."
+      }
+    },
+    "metricName": {
```
Comment (bmoore-msft) [inline monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: add default values for all the metric params

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+        "description": "The percentage to increase the instance count when autoscale engine is initiating scale out action."
+      }
+    },
+    "changePercentScaleIn": {
+      "type": "Int",
+      "metadata": {
+        "description": "The percentage to decrease the instance count when autoscale engine is initiating scale in action."
+      }
+    },
+    "enabled": {
```
Comment (bmoore-msft) [inline monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: rename to something a little less ambiguous - e.g. enableAutoScale

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+      "type": "Bool",
+      "metadata": {
+        "description": "A boolean to indicate whether the autoscale policy is enabled or disabled."
+      }
+    }
+  },
+  "variables": {
+    "settingName": "[concat(toLower(variables('namingInfix')), '-setting')]",
+    "targetResourceId": "[concat('/subscriptions/', subscription().SubscriptionId, '/resourcegroups/', resourceGroup().name, '/providers/Microsoft.Compute/virtualMachineScaleSets/', variables('namingInfix'))]",
```
Comment (bmoore-msft) [inline monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: simplify, just use resourceId('Microsoft.Compute/virtualMachineScaleSets/', variables('namingInfix')))

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+              {
+                "name": "[variables('nicName')]",
+                "properties": {
+                  "primary": "true",
+                  "ipConfigurations": [
+                    {
+                      "name": "[variables('ipConfigName')]",
+                      "properties": {
+                        "subnet": {
+                          "id": "[concat('/subscriptions/', subscription().subscriptionId,'/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'), '/subnets/', variables('subnetName'))]"
```
Comment (bmoore-msft) [inline monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: simplify - just use resource()

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+                        },
+                        "loadBalancerBackendAddressPools": [
+                          {
+                            "id": "[concat('/subscriptions/', subscription().subscriptionId,'/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Network/loadBalancers/', variables('loadBalancerName'), '/backendAddressPools/', variables('bePoolName'))]"
```
Comment (bmoore-msft) [inline monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: simplify - just use resource()

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+                          }
+                        ],
+                        "loadBalancerInboundNatPools": [
+                          {
+                            "id": "[concat('/subscriptions/', subscription().subscriptionId,'/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Network/loadBalancers/', variables('loadBalancerName'), '/inboundNatPools/', variables('natPoolName'))]"
```
Comment (bmoore-msft) [inline monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: simplify - just use resource()

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.parameters.json [hunk 10]
```diff
@@ -0,0 +1,51 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmSku": {
```
Comment (bmoore-msft) [inline monitor-autoscale-vmss-simplemetricbased/azuredeploy.parameters.json]: remove params that have defaultValues

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json (hunk 3 above)
Comment (asheniam) [reply monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: Will fix

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.parameters.json (hunk 10 above)
Comment (asheniam) [reply monitor-autoscale-vmss-simplemetricbased/azuredeploy.parameters.json]: Will fix

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json (hunk 2 above)
Comment (asheniam) [reply monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: Fixed

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json (hunk 1 above)
Comment (asheniam) [reply monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: Fixed

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json (hunk 4 above)
Comment (asheniam) [reply monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: Fixed

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json (hunk 5 above)
Comment (asheniam) [reply monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: Fixed

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json (hunk 6 above)
Comment (asheniam) [reply monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: Fixed

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json (hunk 9 above)
Comment (asheniam) [reply monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: Fixed

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json (hunk 8 above)
Comment (asheniam) [reply monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: Fixed

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json (hunk 7 above)
Comment (asheniam) [reply monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: Fixed

**Comment by asheniam** [comment]
@bmoore-msft - Addressed all CR comments. Please take a look

**Comment by bmoore-msft** [comment]
I'm not seeing any changes - did you update the PR?

**Comment by asheniam** [comment]
@bmoore-msft Yes, I can see the files are updated

File: monitor-autoscale-vmss-simplemetricbased/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,358 @@
...
+      "metadata": {
+        "description": "The Windows version for the VM."
+      }
+    },
+    "vmssName": {
+      "type": "string",
+      "metadata": {
+        "description": "String used as a base for naming resources. Must be 3-61 characters in length and globally unique across Azure."
+      },
+      "minLength": 1,
```
Comment (bmoore-msft) [inline monitor-autoscale-vmss-simplemetricbased/azuredeploy.json]: should be 3?

**Comment by asheniam** [comment]
@bmoore-msft I fixed the final comment on the minLength for VMSS name being incorrect.  Also I apologize in advance -- I've been squashing all my commits so there hasn't been any diff history so this has been hard to review.

**Pull request merged**